sagemaker-featurestore/feature_store_kms_key_encryption.ipynb

```diff
@@ -0,0 +1,593 @@
+    "region = sagemaker_session.boto_region_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create a KMS client using boto3. Note that you can access your boto session through your sagemaker session, e.g.,`sagemaker_session`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kms = sagemaker_session.boto_session.client('kms')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### KMS Policy Template\n",
+    "\n",
+    "Below is the policy template you will use for creating a KMS key. You will specify your role to grant it access to various KMS operations that will be used in the back-end for encrypting your data in your Online or Offline Feature Store."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "policy = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Id\": \"key-policy-feature-store\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Sid\": \"Allow access through Amazon SageMaker Feature Store for all principals in the account that are authorized to use Amazon SageMaker Feature Store\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"AWS\": role\n",
+    "                \n",
+    "            },\n",
+    "            \"Action\": [\n",
+    "                \"kms:Encrypt\",\n",
+    "                \"kms:Decrypt\",\n",
+    "                \"kms:DescribeKey\",\n",
+    "                \"kms:CreateGrant\",\n",
+    "                \"kms:RetireGrant\",\n",
+    "                \"kms:ReEncryptFrom\",\n",
+    "                \"kms:ReEncryptTo\",\n",
+    "                \"kms:GenerateDataKey\",\n",
+    "                \"kms:ListAliases\",\n",
+    "                \"kms:ListGrants\"\n",
+    "            ],\n",
+    "            \"Resource\": [\"*\"],\n",
+    "            \"Condition\": {\n",
+    "                \"StringLike\": {\n",
+    "                    \"kms:ViaService\": \"sagemaker.*.amazonaws.com\"\n",
+    "                }\n",
+    "            }\n",
+    "        },\n",
+    "        {\n",
+    "            \"Sid\": \"Allow administrators to view the CMK and revoke grants\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"AWS\": [role]\n",
+    "            },\n",
+    "            \"Action\": [\n",
+    "                \"kms:Describe*\",\n",
+    "                \"kms:Get*\",\n",
+    "                \"kms:List*\",\n",
+    "                \"kms:RevokeGrant\"\n",
+    "            ],\n",
+    "            \"Resource\": [\"*\"]\n",
+    "        },\n",
+    "        {\n",
+    "            \"Sid\": \"Allow Feature Store to get information about the CMK\",\n",
```

Comment:
FeatureStore impersonates customers to create/get grants via customer's temporary credentials. So We do not need these specifically specified in the policy

Reply:
Removed this policy.